temba/reports/models.py

```diff
@@ -0,0 +1,154 @@
+        return dict(text=self.title, id=self.pk, description=self.description, config=self.config)
+
+    def __str__(self):  # pragma: needs cover
+        return "%s - %s" % (self.pk, self.title)
+
+    class Meta:
+        unique_together = (("org", "title"),)
+
+
+class DataCollectionProcess(models.Model):
+    TYPE_AUTO = "A"
+    TYPE_MANUAL = "M"
+    TYPE_CHOICES = ((TYPE_AUTO, "Automatically"), (TYPE_MANUAL, "Manually"))
+
+    start_type = models.CharField(choices=TYPE_CHOICES, max_length=1, default=TYPE_AUTO)
+    related_org = models.ForeignKey("orgs.Org", on_delete=models.SET_NULL, null=True)
+    started_by = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.PROTECT, null=True)
+    started_on = models.DateTimeField(auto_now_add=True)
+    completed_on = models.DateTimeField(blank=True, null=True)
+    flows_total = models.PositiveIntegerField(default=0)
+    flows_skipped = models.PositiveIntegerField(default=0)
+    flows_processed = models.PositiveIntegerField(default=0)
+
+    @classmethod
+    def get_last_collection_process_status(cls, org):
+        last_dc: DataCollectionProcess = (
+            cls.objects.filter(Q(related_org=org) | Q(start_type=cls.TYPE_AUTO)).order_by("-started_on").first()
+        )
+        if last_dc:
+            data_status = {
+                "lastUpdated": str(last_dc.started_on),
+                "completed": bool(last_dc.completed_on),
+                "progress": (
+                    (last_dc.flows_skipped + last_dc.flows_processed) / last_dc.flows_total
+                    if last_dc.flows_total
+                    else 1
+                ),
+            }
+        else:
+            data_status = {"lastUpdated": None, "completed": True, "progress": 1}
+        return data_status
+
+
+class CollectedFlowResultsData(models.Model):
+    """
+    A model to store flow results data that is required for analytics reports, in more practical way.
+    {
+        result_key: {
+            # ids of all contacts answered with that category
+            category_key: [1, 4, 15],
+            another_category_key: [3, 7]
+        }
+    }
+    """
+
+    flow = models.OneToOneField("flows.Flow", on_delete=models.PROTECT, related_name="aggregated_results")
+    data = JSONField(default=dict)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def collect_results_data(cls, flow: Flow):
+        flow_results = {result["key"]: result["categories"] for result in flow.metadata["results"]}
+        result_keys = sorted(list(flow_results.keys()))
+        is_data_exists = cls.objects.filter(flow_id=flow.id).exists()
+        is_data_already_updated = is_data_exists and not flow.runs.filter(
+            modified_on__gt=flow.aggregated_results.last_updated
+        )
+        if is_data_already_updated or not result_keys:
+            return
+
+        # select runs data and build the aggregated data to save
+        final_data = defaultdict(lambda: defaultdict(set))
```

Comment:
I think would be better to have the contacts count not using unique contact ids, as it could be a little confusing for users, that is, not being unique they will be able to see the same number (or closer to it) on the flow editor and also on the analytics.

Reply:
Have replaced set logic with list logic. And fixed the issue with the refresh data window state.